_pages/algorithm-development/source-control-management.md

```diff
@@ -428,3 +434,62 @@ Yes! Upon any change to your repository's name or owner we receive a webhook whi
 __I revoked access to the Algorithmia's GitLab OAuth application, and now none of the repositories I created can build!__
 
 When you revoke access to an OAuth app, GitLab automatically revokes both your token and any deploy keys you may have created via the OAuth app. To fix your algorithms, simply [follow the instructions above](#your-repositorys-deploy-key-was-removed) on restoring deploy keys to a repository.
+
+
+### Hosting Source Code on Bitbucket Server
+**No Oauth Support** Bitbucket Server does not support OAuth, so a token based flow is implemented to connect Bitbucket Server-hosted algorithms. 
+{: .notice-info}
+
+By hosting your algorithm's source code on Bitbucket Server, you can take advantage of Bitbucket Server's rich set of developer features, such as pull requests, and also ensure that access to your source code is carefully mediated.
+
+If you're new to Git, we recommend [this tutorial series](https://try.github.io/) by GitHub.
+
+**Web IDE Support** At this time we do not support editing source code in our web app for Bitbucket Server-hosted algorithms. 
+{: .notice-info}
+
+
+#### Creating Algorithms with Bitbucket Server
+
+To host your algorithm on Bitbucket Server, simply select the Bitbucket Server instance you wish to use when creating your algorithm.
+
+Once you've selected your Bitbucket Server instance, you will be able to host algorithm source code in Bitbucket Server.
+
+You can customize two aspects of the Bitbucket Server repositories that are created for your algorithms: the repository's owner, and its name.
```

Comment:
change to:
You can customize two aspects of any Bitbucket Server repository that's created for an algorithm: ...